### PR TITLE
Fix version comparison issue in compatibility check stage

### PR DESCRIPTION
**Pull request by zhangmeng916**
### Issues

- [X] My PR addresses the following Helix issues and references them in the PR description:

(#990 )

### Description

- [X] Here are some details about my PR, including screenshots of any UI changes:
Currently there is a bug in the version comparison logic for compatibility check stage. It worked with all version starting with number "0". However, the comparison check would fail after we release version 1.0.0. The impact is the pipeline that includes compatibility check stage would fail due to the incompatibility error.
This PR fixed the issue by modifying the logic in the loop. If the participant version's digit number is larger than the minimum version digit number, we terminate the loop.

### Tests

- [X] The following tests are written for this issue:

TestCompatibilityCheckStage.

- [X] The following is the result of the "mvn test" command on the appropriate module:
helix-rest
[INFO] Tests run: 159, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 32.061 s - in TestSuite
[INFO] 
[INFO] Results:
[INFO] 
[INFO] Tests run: 159, Failures: 0, Errors: 0, Skipped: 0
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  36.217 s
[INFO] Finished at: 2020-05-03T12:01:15-07:00
[INFO] ------------------------------------------------------------------------

helix-core:
[ERROR] Tests run: 1144, Failures: 1, Errors: 0, Skipped: 1, Time elapsed: 4,742.047 s <<< FAILURE! - in TestSuite
[ERROR] testJobQueueAutoCleanUp(org.apache.helix.integration.task.TestJobQueueCleanUp)  Time elapsed: 300.01 s  <<< FAILURE!
org.testng.internal.thread.ThreadTimeoutException: Method org.testng.internal.TestNGMethod.testJobQueueAutoCleanUp() didn't finish within the time-out 300000

[INFO] 
[INFO] Results:
[INFO] 
[ERROR] Failures: 
[ERROR]   TestJobQueueCleanUp.testJobQueueAutoCleanUp » ThreadTimeout Method org.testng....
[INFO] 
[ERROR] Tests run: 1144, Failures: 1, Errors: 0, Skipped: 1
### Commits

- [X] My commits all reference appropriate Apache Helix GitHub issues in their subject lines. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Code Quality

- [X] My diff has been formatted using helix-style.xml 
(helix-style-intellij.xml if IntelliJ IDE is used)

### CODE REVIEW

File: helix-core/src/main/java/org/apache/helix/controller/stages/CompatibilityCheckStage.java [hunk 1]
```diff
@@ -55,7 +55,7 @@ public void process(ClusterEvent event) throws Exception {
             "incompatible participant. pipeline will not continue. " + "controller: "
                 + manager.getInstanceName() + ", controllerVersion: " + properties.getVersion()
                 + ", minimumSupportedParticipantVersion: "
-                + properties.getProperty("miminum_supported_version.participant")
+                + properties.getProperty("minimum_supported_version.participant")
```
Comment (dasahcc) [inline helix-core/src/main/java/org/apache/helix/controller/stages/CompatibilityCheckStage.java:58]: What's the difference here?

**Comment by zhangmeng916** [comment]
This PR is ready to merge, approved by @dasahcc 
Final commit message "Fix version comparison issue in compatibility check stage. Make sure new version number will not fail this check".

File: helix-core/src/main/java/org/apache/helix/controller/stages/CompatibilityCheckStage.java (hunk 1 above)
Comment (zhangmeng916) [reply helix-core/src/main/java/org/apache/helix/controller/stages/CompatibilityCheckStage.java:58]: The previous one has a typo.

Comment (zhangmeng916) [reply helix-core/src/main/java/org/apache/helix/controller/stages/CompatibilityCheckStage.java:58]: And it caused the log to always show "null" for minimum support version, as it cannot find it in the property file. So the log is very confusing during debug.

File: helix-core/src/main/java/org/apache/helix/HelixManagerProperties.java [hunk 2]
```diff
@@ -116,6 +116,8 @@ static boolean versionNoLessThan(String version1, String version2) {
 
         if (versionNum1 < versionNum2) {
           return false;
+        } else if (versionNum1 > versionNum2) {
```
Comment (lei-xia) [inline helix-core/src/main/java/org/apache/helix/HelixManagerProperties.java:119]: This just ignore the check, right? Can we make the logic correct here, say 1.0.x > 0.9.x?

Comment (zhangmeng916) [reply helix-core/src/main/java/org/apache/helix/HelixManagerProperties.java:119]: This is not to ignore the check. This is a digit by digit comparison. See above code, the version is already split. So we are comparing "1" to "0", which breaks the loop. If it's "0" to "0", we will continue to compare next digit. It should be complete.

Comment (zhangmeng916) [reply helix-core/src/main/java/org/apache/helix/HelixManagerProperties.java:119]: Previous error is just that it didn't break even the digit is larger the minimum version. I guess it didn't consider we might bump to 1.0.0.0.

**Pull request merged**